pandas/_libs/lib.pyx

```diff
@@ -1210,6 +1210,28 @@ try:
 except AttributeError:
     pass
 
+cdef enum CurrSeen:
+    bool_t = 0
+    null_t = 1
+    uint_t = 2
+    int_t = 3
+    float_t = 4
+    complex_t = 5
+
+cdef union CurrSeenVal:
+    uint8_t bool_t
+    uint64_t uint_t
+    int64_t int_t
+    float64_t float_t
```

Comment:
i dont think we use bool_t, uint_t, int_t, float_t elsewhere.  can we avoid these?